fe/fe-core/src/main/java/com/starrocks/sql/common/TypeManager.java

```diff
@@ -121,7 +121,13 @@ public static Type getCompatibleTypeForBetweenAndIn(List<Type> types) {
         return compatibleType;
     }
 
-    public static Type getCompatibleTypeForBinary(Type type1, Type type2) {
+    public static Type getCompatibleTypeForBinary(boolean isEquivalence, Type type1, Type type2) {
+        if (isEquivalence) {
+            if ((type1.isStringType() && type2.isNumericType()) || (type1.isNumericType() && type2.isStringType())) {
```

Comment:
would better comment 1： why we only handle "=" operator? 2 why the comment type is string?